### PR TITLE
LG-3108: modal styles updates

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-3108](https://jira.mongodb.org/browse/LG-3108)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

**Comment by bruugey** [comment]
Spoke to @audroue and we actually think in this case it makes sense to just remove the p-specific styling from dark mode. Instead, let's keep the theme defaults as they are and allow consuming applications to override this as they wish by providing a className to whatever typography elements they are passing to Modal components as children. Let me know if you disagree @DesignerDave

**Comment by DesignerDave** [comment]
@brookeyalof I think keeping a sensible default for the mode is something we've heard a number of asks for in the past, and I believe we do this for components like Tooltip.

Also worth noting that I would consider it a breaking change to remove a default styling for dark mode. Existing dark mode modals would be, in theory, broken with the new version if they relied on this styling.

What problem are we solving by removing this?

**Comment by DesignerDave** [comment]
@bruugey We have a pattern for making changes to components within others now. In Popover, for example, we use a context to change how the component behaves based on being rendered in a Modal. We should probably be setting a default using a similar approach, or generally be changing the default color for our Paragraph component based on the LeafyGreenProvider value itself.

Either way, this would be a breaking change if we pursue either approach.

**Comment by bruugey** [comment]
Sounds like what you're talking about @DesignerDave is adaptive defaults (our components smartly rendering themselves based on their context). We have a few tickets that get at this (callouts being placed after headers, tables having proper padding when rendered inside of cards). The scope of that work feels to big for this ticket here. I think either we can remove the p specific styles as we're still assigning the bounding `<div>{children}</div>` or explore the solution proposed in this PR, but I do think that assigning all inheriting p tags a specific color is not the right go forward.

File: .changeset/angry-teachers-scream.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/modal': patch
```
Comment (bruugey) [inline .changeset/angry-teachers-scream.md:2]: breaking

File: .changeset/angry-teachers-scream.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@leafygreen-ui/modal': major
+---
+
+Removes the default `P` styles applied in darkMode that used to override the color of all `P` tags with `#C1C7C6`. Child element colors are now respected and if no colors are explicitly defined for them, they will inherit the color of their parent container. Also reorganizes internal file structure.
```
Comment (bruugey) [inline .changeset/angry-teachers-scream.md:5]: nit: lc p

File: packages/modal/package.json [hunk 3]
```diff
@@ -36,7 +36,7 @@
   "devDependencies": {
     "@leafygreen-ui/select": "^10.3.1",
     "@leafygreen-ui/button": "^20.0.4",
-    "@leafygreen-ui/typography": "^16.2.1",
+    "@leafygreen-ui/typography": "^16.3.0",
```
Comment (bruugey) [inline packages/modal/package.json:39]: why do we have to manually version bump this?

**Review by bruugey — APPROVED** [review]

**Pull request merged**